_posts/2019-03-12-so-you-want-to-be-an-engineer.md

```diff
@@ -0,0 +1,170 @@
+TypeScript code in a React framework?"
+
+Fortunately, a lot of other people have the same question.
+
+# Step Three: Decide How You Want to Learn It
+
+First: **there is no wrong way to go about learning to code**. Whether it takes you 6 months or 6 years, whether
+you learn one language or a dozen, whether you ask for a lot of help or do it all yourself, **you are learning and
+that is valuable**. Learning to code is not a magical skill. Like just about everything, it's a matter of putting
+in time. Just keep trying, even small things, and you _will_ make progress.
+
+Lots of people who want to become engineers go to coding bootcamps like [Flatiron](https://flatironschool.com/) or
+[General Assembly](https://generalassemb.ly/). That's a very reasonable thing to do! I had a decent grounding in CS
+fundamentals thanks to my experience in undergrad, and as a result, I initially decided I wouldn't do a coding
+bootcamp—I felt I had enough experience to benefit from the multitude of online courses and open source projects
+out there.
+
+However, there are definitely real and significant benefits do doing a bootcamp. Here are three that I can think
+of:
+
+1. Clear curriculum. At times, I was overwhelmed by the possible ways to proceed and stopped making progress as a
+   result. Bootcamps take the guesswork out.
+2. Timeline. Because I didn't have a hard date set for a transition or interview, it was up to me to determine how
+   fast/slow I worked on things, and I stagnated at times as a result.
+3. Community. While the open source community is very much a real thing, and you can find lovely people on the
+   internet who want to help learners like yourself, it's not the same as having multiple people learning the same
+   thing at the same time in the same physical space.
+
+The downsides, of course, are that coding bootcamps are expensive and time-consuming. Most of us don't have the
+luxury to leave a job, pay $10K – $15K in fees, and spend a couple months at a full-time bootcamp with no income.
+There are other models, such as online-only bootcamps or after-hours classes, but those come with challenges of
+their own.
+
+I'd say that If you can afford a bootcamp and have the time, it's a great way to jumpstart a coding career, but you
+don't have to attend one to become a good engineer.
+
+If you decide to go the non-bootcamp route, you also have a lot of good options. I did courses through
+[Udemy](https://www.udemy.com), which has a lot of courses that are literally always on sale for $10 - $20. There
+are many other similar services out there as well, like [Udacity](https://www.udacity.com),
+[Codecademy](https://www.codecademy.com/), and [Treehouse](https://teamtreehouse.com/).
+
+I also highly recommend attending IRL meetups, because that's where you can make connections and learn from others
+most easily. [Meetup.com](https://www.meetup.com/find/tech/) is a great place to start.
```

Comment:
If you found any specific meetups particularly helpful ([hint hint](https://peerlab.community)) it'd be cool to link to them here.

Reply:
good call! updated!